api/v1beta1/condition_messages.go

```diff
@@ -4,31 +4,28 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type ConditionBuilder struct {
-	Status             metav1.ConditionStatus
-	Reason             KymaConditionReason
-	ObservedGeneration int64
-}
-
 const (
 	MessageModuleInReadyState       = "all modules are in ready state"
 	MessageModuleNotInReadyState    = "not all modules are in ready state"
 	MessageModuleCatalogIsSynced    = "module catalog is synchronized"
 	MessageModuleCatalogIsOutOfSync = "module catalog is out of sync and needs to be resynchronized"
 	MessageSKRWebhookIsSynced       = "skrwebhook is synchronized"
 	MessageSKRWebhookIsOutOfSync    = "skrwebhook is out of sync and needs to be resynchronized"
+
+	ReasonReady      = "Ready"
+	ReasonProcessing = "Processing"
 )
 
 // Extend this list by actual needs.
 const (
-	ConditionReasonModulesAreReady      KymaConditionReason = "ModulesAreReady"
-	ConditionReasonModuleCatalogIsReady KymaConditionReason = "ModuleCatalogIsReady"
-	ConditionReasonSKRWebhookIsReady    KymaConditionReason = "SKRWebhookIsReady"
+	ConditionTypeModulesAreReady      KymaConditionType = "ModulesAreReady"
+	ConditionTypeModuleCatalogIsReady KymaConditionType = "ModuleCatalogIsReady"
+	ConditionTypeSKRWebhookIsReady    KymaConditionType = "SKRWebhookIsReady"
 )
 
-func GenerateMessage(reason KymaConditionReason, status metav1.ConditionStatus) string {
-	switch reason {
-	case ConditionReasonModulesAreReady:
+func GenerateMessage(conditionType KymaConditionType, status metav1.ConditionStatus) string {
```

Comment:
Just left my proposal here for reference:
IMO, this `GenerateMessage` function is fine, we don't need to change it, the logic here is a bit complicate I agree, but it's designed solid, and it's not this generate messge cause problem.
The root cause of the current problem is we don't have different conditionType.
Why not just design three conditiontype one-to-one mapped with reason?

for example: ConditionTypeModules (`Modules`) <-> ConditionReasonModulesAreReady (`ModulesAreReady`)
Then in the existing `UpdateCondition`, we also don't need to change any signatures, still use `reason KymaConditionReason`, the only change need to do is just add a function to convert reason to type.
```
Type:               convertToType(reason)
```

Reply:
Honestly imo the entire file is pretty much over engineered and we would be fine with something like this:

```suggestion

type ConditionType string

const (
	ConditionTypeModules    ConditionType = "Modules"
)

type ConditionReason string

const (
	ConditionReasonModulesAvailable ConditionReason = "Available"
)

type ConditionMessage string

const (
       ConditionMessageModulesAvailable ConditionMessage = "modules are synchronised and available"
)


func newResourcesCondition(obj metav1.Object) metav1.Condition {
	return metav1.Condition{
		Type:               string(ConditionTypeModules),
		Reason:             string(ConditionReasonModulesAvailable),
		Message:            string(ConditionMessageModulesAvailable),
		Status:             metav1.ConditionFalse,
		ObservedGeneration: obj.GetGeneration(),
	}
}
```

Reply:
I agree with @ruanxin at this point.
We had a couple of discussion and we agreed on the following format:
```
metav1.Condition{
	Type:               Can be `Modules`, `ModuleCatalogue`, `SKRWebhook`
	Status:             True, False or Unknown,
	Reason:             `Ready`,
	Message:            Message which actually represents the current state in human readable format,
	ObservedGeneration: kyma.GetGeneration(),
}
```